retries.go

```diff
@@ -54,7 +57,19 @@ func tooManyRequestsRetryCondition(r *resty.Response, _ error) bool {
 }
 
 func serviceUnavailableRetryCondition(r *resty.Response, _ error) bool {
-	return r.StatusCode() == http.StatusServiceUnavailable
+	var retry bool
+	if r.StatusCode() == http.StatusServiceUnavailable {
+		// During maintenance events, the API will return a 503 and add
+		// an `X-MAINTENANCE-MODE` header. Don't rety during maintenance
+		// events, only for legitimate 503s.
+		if r.Header().Get(maintenanceModeHeaderName) != "" {
+			log.Printf("[INFO] Linode API is under maintenance, request will not be retried")
+		} else {
+			retry = true
+		}
+	}
+
+	return retry
```

Comment:
We might be able to simplify the logic here a bit :+1: 

```suggestion
	serviceUnavailable := r.StatusCode() == http.StatusServiceUnavailable

	// During maintenance events, the API will return a 503 and add
	// an `X-MAINTENANCE-MODE` header. Don't retry during maintenance
	// events, only for legitimate 503s.
	if serviceUnavailable && r.Header().Get(maintenanceModeHeaderName) != "" {
		log.Printf("[INFO] Linode API is under maintenance, request will not be retried")
		return false
	}

	return serviceUnavailable
```

Reply:
Fine by me! I don't mind switching to inline conditionals, they're more prevalent in this bit of code anyway. 👍